File: cs/test/presentation.cs
// Copyright (c) DraviaVemal. Licensed under the MIT License. See License in the project root.

using draviavemal.openxml_office.global_2007;
using draviavemal.openxml_office.presentation_2007;

namespace openxmloffice.tests
{
    /// <summary>
    /// Power Point Test
    /// </summary>
    [TestClass]
    public class Presentation
    {
        private static readonly string resultPath = "../../test_results";
        /// <summary>
        /// Initialize power point Test
        /// </summary>
        /// <param name="context">
        /// </param>
        [ClassInitialize]
        public static void ClassInitialize(TestContext context)
        {
            if (!Directory.Exists(resultPath))
            {
                Directory.CreateDirectory(resultPath);
            }
            PrivacyProperties.ShareComponentRelatedDetails = false;
            PrivacyProperties.ShareIpGeoLocation = false;
            PrivacyProperties.ShareOsDetails = false;
            PrivacyProperties.SharePackageRelatedDetails = false;
            PrivacyProperties.ShareUsageCounterDetails = false;
        }
        /// <summary>
        /// Save the Test File After execution
        /// </summary>
        [ClassCleanup]
        public static void ClassCleanup()
        {
            // powerPoint.SaveAs(string.Format("{1}/test-{0}.xlsx", DateTime.Now.ToString("yyyy-MM-dd-HH-mm-ss"), resultPath));
        }

        /// <summary>
        /// 
        /// </summary>
        // [TestMethod]
        public void BlankFile()
        {
            PowerPoint powerPoint2 = new(new()
            {
                IsInMemory = false
            });
            powerPoint2.SaveAs(string.Format("{1}/Blank-{0}.pptx", DateTime.Now.ToString("yyyy-MM-dd-HH-mm-ss"), resultPath));
        }

        /// <summary>
        /// Test existing file
        /// </summary>
        [TestMethod]
        public void OpenExistingPowerPoint()
        {
            PowerPoint powerPoint1 = new("test_files/basic_test.pptx", new()
            {
                IsInMemory = false
            });
            powerPoint1.SaveAs(string.Format("{1}/EditStyle-{0}.pptx", DateTime.Now.ToString("yyyy-MM-dd-HH-mm-ss"), resultPath));
            Assert.IsTrue(true);
        }
    }
}

File: test/rs/src/presentation.rs
#[cfg(test)]
mod presentation_test {
    use chrono::Utc;
    use draviavemal_openxml_office::presentation_2007::{PowerPoint, PowerPointPropertiesModel};
    use std::fs::{create_dir_all, exists};

    fn get_save_file(dynamic_path: Option<&str>) -> String {
        let result_path = "test_results";
        if !exists(result_path).expect("Dir Check Failed") {
            create_dir_all(result_path).expect("Failed to Create")
        }
        format!(
            "{}/test-{}{}.pptx",
            result_path,
            dynamic_path.unwrap_or(""),
            Utc::now().format("%Y-%m-%d-%H-%M-%S-%3f").to_string()
        )
    }

    #[test]
    fn blank_power_point() {
        let file = PowerPoint::new(None, PowerPointPropertiesModel::default())
            .expect("Create New File Failed");
        file.save_as(&get_save_file(None))
            .expect("Failed to save Empty Power Point");
        assert_eq!(true, true);
    }

    #[test]
    fn edit_power_point() {
        let file = PowerPoint::new(
            Some("src/TestFiles/basic_test.pptx".to_string()),
            PowerPointPropertiesModel::default(),
        )
        .expect("Open Existing file failed");
        file.save_as(&get_save_file(None))
            .expect("Failed to save Edit Power Point");
        assert_eq!(true, true);
    }
}
